Add cmdref generated documentation for `clustermesh-apiserver`

HadrienPatte opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

---

This PR updates the `clustermesh-apiserver` CLI to allow it to generate its cmdref and updates the documentation with that newly generated cmdref.

---

```release-note
Add cmdref generated documentation for clustermesh-apiserver
```

HadrienPatte committed:
clustermesh-apiserver: update CLI to generate cmdref

Signed-off-by: Hadrien Patte <hadrien.patte@datadoghq.com>

HadrienPatte committed:
clustermesh-apiserver: generate cmdref documentation

Generated by running clustermesh-apiserver --cmdref

Signed-off-by: Hadrien Patte <hadrien.patte@datadoghq.com>

giorio94 commented on clustermesh-apiserver/main.go:
```diff
@@ -18,13 +19,21 @@ import (
+	flags := rootCmd.PersistentFlags()
+	flags.StringVar(&cmdRefDir, "cmdref", "", "Path to cmdref output directory")
+	flags.MarkHidden("cmdref")
+}
```
Is there any reason for controlling this functionality via a flag, rather than a dedicated `cmdref` command? That looks more idiomatic and less error-prone to me. 

Ideally, all binaries could also reuse a common `cmdref` command ([here](https://github.com/cilium/cilium/blob/main/daemon/cmd/root.go#L51-L59) is the agent one, for instance), to ensure consistency and reduce code duplication, but that can definitely be a follow-up.

HadrienPatte replied:
I mapped this implementation on the one from [cilium-health](https://github.com/cilium/cilium/blob/ca18209dd67a5b6e8e3cf6b114441cc9495d37e3/cilium-health/cmd/root.go#L64), which uses a flag, I can update it to use a subcommand instead.

HadrienPatte replied:
Updated in [0dad7b5](https://github.com/cilium/cilium/pull/36205/commits/0dad7b5b018ef1857124b27eaf9dbe9350a8236c)

giorio94 commented on clustermesh-apiserver/main.go:
```diff
@@ -35,8 +44,28 @@ func main() {
+		if err := doc.GenMarkdownTreeCustom(rootCmd, cmdRefDir, filePrepend, linkHandler); err != nil {
+			fmt.Println(err)
+		}
+		os.Exit(0)
```
~~This doesn't seem to work, because the execution would terminate with an error after `rootCmd.Execute` above when running `clustermesh-apiserver --cmdref <path>`.~~

Edit: the docs are actually generated (because `rootCmd.Execute` does not return an error in that case), but the command emits the usage prompt, which is misleading.

giorio94 commented on clustermesh-apiserver/main.go:
```diff
@@ -18,13 +19,21 @@ import (
+var rootCmd = &cobra.Command{
+	Use:   "clustermesh-apiserver",
+	Short: "Run the ClusterMesh apiserver",
+}
```
This change doesn't look necessary to implement this functionality.

giorio94 reviewed (changes requested):
Thanks! A couple of comments inline.

giorio94 reviewed (approved):
Looks great to me, thanks! 
Could you please just squash the new commits into the old ones?

HadrienPatte committed:
clustermesh-apiserver: fix spelling

Fix "Initialise" spelling in clustermesh-apiserver etcdinit command and
add "etcdinit" to spelling wordlist.

See https://github.com/cilium/cilium/actions/runs/12054803453/job/33614531375?pr=36205

Signed-off-by: Hadrien Patte <hadrien.patte@datadoghq.com>

HadrienPatte committed:
clustermesh-apiserver: add clustermesh-apiserver to main Makefile

This is necessary to ensure that the clustermesh-apiserver binary gets
built in order to generate its cmdref documentation.

See https://github.com/cilium/cilium/actions/runs/12054803453/job/33614523855?pr=36205

Signed-off-by: Hadrien Patte <hadrien.patte@datadoghq.com>
